README.txt
======================================================
Base module for Nagios check plugins written in Python
======================================================

This package provides a family of Python modules
to streamline writing Nagios plugins.

These modules are rewritten modules originated by the
Perl module family Nagios::Plugin.

Following modules are supported:

 - nagios
   - classes:
     - BaseNagiosError
   - decorators:
     - constant
   - objects:
     - state
       - readonly properties:
         - ok
         - warning
         - critical
         - unknown
         - dependent
 - nagios.plugin
   - classes:
     - NagiosPlugin
     - NagiosPluginError
 - nagios.plugin.config
   - classes:
     - NagiosPluginConfig
 - nagios.plugin.functions
   - functions:
     - nagios_exit()
     - nagios_die()
     - check_messages()
     - get_shortname()
     - max_state()
     - max_state_alt()
 - nagios.plugin.getopt
     - classes:
       - NagiosPluginGetoptParser
 - nagion.plugin.perf
     - classes:
       - NagiosPluginPerformance
 - nagion.plugin.range
     - classes:
       - NagiosPluginRange
 - nagios.plugin.threshold
     - classes:
       - NagiosPluginThreshold


Author: Frank Brehm (<frank.brehm@profitbricks.com>)


debian/changelog
pb-monitoring-plugins (1.9.6) all; urgency=medium

  * check_ssd_lifetime: use S.M.A.R.T value 233 (Redmine: #27096)

 -- Tobias von der Krone <tobias.vonderkrone@profitbricks.com>  Wed, 19 Oct 2016 11:39:22 +0200

pb-monitoring-plugins (1.9.5) all; urgency=medium

  * check_ssd_lifetime: some fixes (Redmine: #27096)
    - fix for threshold comparision
    - fix for parsing smartctl output with , and . with us and de language
    - fix for echoing message when state is warning or critical

 -- Thomas Moericke <thomas.moericke@profitbricks.com>  Fri, 14 Oct 2016 15:01:34 +0200

pb-monitoring-plugins (1.9.4) all; urgency=medium

  * check_pb_consistence_storage: enforce root permissions (Redmine: #26775)

 -- Thomas Moericke <thomas.moericke@profitbricks.com>  Fri, 30 Sep 2016 08:01:58 +0200

pb-monitoring-plugins (1.9.3) all; urgency=medium

  * check_ssd_lifetime: fix check for controller bus number (#20486)

 -- Agata Nazar <agata.nazar@profitbricks.com>  Wed, 24 Aug 2016 14:09:17 +0200

pb-monitoring-plugins (1.9.2) all; urgency=medium

  * pb-monitoring-plugins: Depend on libswitch-perl (required by
    check-multipath.pl)

 -- Benjamin Drung <benjamin.drung@profitbricks.com>  Fri, 12 Aug 2016 18:57:42 +0200

pb-monitoring-plugins (1.9.1) all; urgency=medium

  * check_backup_lvm_push: Fix check alerts "Timestamps differ"
    (Redmine: #24503)

 -- Jan Hendrick Reiche <jan-hendrick.reiche@profitbricks.com>  Wed, 13 Jul 2016 16:12:05 +0200

pb-monitoring-plugins (1.9.0) all; urgency=medium

  [ Benjamin Drung ]
  * Fix flake8 errors in many script (and fix some bugs found by flake8)
  * check_vcb_instance and check_ppd_instance: Fix version comparison
    (Redmine: #23796)
  * check_uname: Also use Debian version comparison for kernel version
  * Enforce flake8 check on build time

  [ Frank Brehm ]
  * Apply flake8 rules to check_pb_storage_exports.py and nagios/plugin/*.py
  * check_softwareraid: Change sorting of found MD devices

 -- Benjamin Drung <benjamin.drung@profitbricks.com>  Fri, 24 Jun 2016 12:44:41 +0200

pb-monitoring-plugins (1.8.32) all; urgency=medium

  * Change in cache configuration for OS VD in check "Cache RAID status"

 -- Agata Nazar <agata.nazar@profitbricks.com>  Wed, 22 Jun 2016 10:53:19 +0200

pb-monitoring-plugins (1.8.31) all; urgency=medium

  * Fix output for check "Cache RAID status"

 -- Agata Nazar <agata.nazar@profitbricks.com>  Thu, 26 May 2016 18:04:56 +0200

pb-monitoring-plugins (1.8.30) all; urgency=medium

  * Change internal cmd_run to support output redirection

 -- Thomas Kroll <thomas.kroll@profitbricks.com>  Mon, 23 May 2016 11:06:27 +0200

pb-monitoring-plugins (1.8.29) all; urgency=medium

  * Change regexp for find birdc6 gw distribution

 -- Thomas Kroll <thomas.kroll@profitbricks.com>  Fri, 20 May 2016 14:45:18 +0200

pb-monitoring-plugins (1.8.28) all; urgency=medium

  * Change the way birdc6 is used and also fixed some minor issues. #18123

 -- Thomas Kroll <thomas.kroll@profitbricks.com>  Thu, 19 May 2016 11:16:22 +0200

pb-monitoring-plugins (1.8.27) all; urgency=medium

  * Fix multipath typo 

 -- Agata Nazar <agata.nazar@profitbricks.com>  Wed, 13 Apr 2016 15:01:40 +0200

pb-monitoring-plugins (1.8.26) all; urgency=medium

  * Fix in check-multipath.pl to remove problems with multipathd socket connection. #21043

 -- Agata Nazar <agata.nazar@profitbricks.com>  Wed, 13 Apr 2016 13:58:39 +0200

pb-monitoring-plugins (1.8.25) all; urgency=medium

  * Add sudo permission for new script check_ssd_lifetime
  * Run wrap-and-sort

 -- Benjamin Drung <benjamin.drung@profitbricks.com>  Tue, 12 Apr 2016 16:49:41 +0200

pb-monitoring-plugins (1.8.24) all; urgency=medium

  * Change in check-multipath.pl multipathd handling #21113

 -- Agata Nazar <agata.nazar@profitbricks.com>  Fri, 08 Apr 2016 15:10:10 +0200

pb-monitoring-plugins (1.8.23) all; urgency=medium

  * Add check check-multipath.pl and sudo permission. #21113

 -- Agata Nazar <agata.nazar@profitbricks.com>  Thu, 07 Apr 2016 14:04:59 +0200

pb-monitoring-plugins (1.8.22) all; urgency=low

  * Don't fetch expensive contract data for storage consistency checks (redmine #21247)

 -- Ingo Fischer <ingo.fischer@profitbricks.com>  Thu, 24 Mar 2016 16:03:50 +0100

pb-monitoring-plugins (1.8.21) all; urgency=medium

  * fix in check_ipmi_sensor

 -- Agata Nazar <agata.nazar@profitbricks.com>  Fri, 18 Mar 2016 17:05:13 +0100

pb-monitoring-plugins (1.8.20) all; urgency=medium

  * update check_ipmi_sensor to version 3.10 #20963

 -- Agata Nazar <agata.nazar@profitbricks.com>  Thu, 17 Mar 2016 16:43:55 +0100

pb-monitoring-plugins (1.8.19) all; urgency=medium

  * Add sudo permission for new scripts check_backup_lvm_push,
    check_cache_raid_status, and check_bgp_propagation (Redmine: #11233)

 -- Benjamin Drung <benjamin.drung@profitbricks.com>  Thu, 10 Mar 2016 11:47:12 +0100

pb-monitoring-plugins (1.8.18) all; urgency=medium

  * Fix for bugs in check_cache_raid_status #19697

 -- Agata Nazar <agata.nazar@profitbricks.com>  Tue, 08 Mar 2016 18:05:37 +0100

pb-monitoring-plugins (1.8.17) all; urgency=medium

  * add new plugin check_cache_raid_status #19697

 -- Agata Nazar <agata.nazar@profitbricks.com>  Thu, 03 Mar 2016 15:45:53 +0100

pb-monitoring-plugins (1.8.16) all; urgency=medium

  [ Thomas Kroll ]
  * add new plugin check_bgp_propagation

  [ Tobias von der Krone ]
  * add new plugin check_ssd_lifetime

 -- Tobias von der Krone <tobias.vonderkrone@profitbricks.com>  Wed, 24 Feb 2016 15:22:37 +0100

pb-monitoring-plugins (1.8.15) all; urgency=medium

  * ajust Output to one line and correct critical timestamp message - #20086

 -- Jan Hendrick Reiche <jan-hendrick.reiche@profitbricks.com>  Wed, 10 Feb 2016 18:09:04 +0100

pb-monitoring-plugins (1.8.14) all; urgency=medium

  * Enhance check for backup-lvm checking if git push was made. - #20086

 -- Jan Hendrick Reiche <jan-hendrick.reiche@profitbricks.com>  Wed, 10 Feb 2016 13:11:41 +0100


nagios/color_syslog.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
"""
@author: Frank Brehm
@contact: frank.brehm@profitbricks.com
@organization: Profitbricks GmbH
@copyright: © 2010 - 2015 by Profitbricks GmbH
@license: GPL3
@summary: module for a colored formatter for syslog
          plugin framework
"""

import logging
import copy


# =============================================================================
# module variables and helper functions

COLOR_CODE = {
    'ENDC':              0,  # RESET COLOR
    'BOLD':              1,
    'UNDERLINE':         4,
    'BLINK':             5,
    'INVERT':            7,
    'CONCEALD':          8,
    'STRIKE':            9,
    'GREY30':           90,
    'GREY40':            2,
    'GREY65':           37,
    'GREY70':           97,
    'GREY20_BG':        40,
    'GREY33_BG':       100,
    'GREY80_BG':        47,
    'GREY93_BG':       107,
    'DARK_RED':         31,
    'RED':              91,
    'RED_BG':           41,
    'LIGHT_RED_BG':    101,
    'DARK_YELLOW':      33,
    'YELLOW':           93,
    'YELLOW_BG':        43,
    'LIGHT_YELLOW_BG': 103,
    'DARK_BLUE':        34,
    'BLUE':             94,
    'BLUE_BG':          44,
    'LIGHT_BLUE_BG':   104,
    'DARK_MAGENTA':     35,
    'PURPLE':           95,
    'MAGENTA_BG':       45,
    'LIGHT_PURPLE_BG': 105,
    'DARK_CYAN':        36,
    'AUQA':             96,
    'CYAN_BG':          46,
    'LIGHT_AUQA_BG':   106,
    'DARK_GREEN':       32,
    'GREEN':            92,
    'GREEN_BG':         42,
    'LIGHT_GREEN_BG':  102,
    'BLACK':            30,
}


# -----------------------------------------------------------------------------
def termcode(num):

    return '\033[%sm' % (num)


# -----------------------------------------------------------------------------
def colorstr(message, color):
    """
    Wrapper function to colorize the message.

    @param message: The message to colorize
    @type message: str
    @param color: The color to use, must be one of the keys of COLOR_CODE
    @type color: str

    @return: the colorized message
    @rtype: str

    """

    tcode = ''
    if isinstance(color, (list, tuple)):
        for c in color:
            tcode += termcode(COLOR_CODE[c])
    else:
        tcode = termcode(COLOR_CODE[color])

    return tcode + message + termcode(COLOR_CODE['ENDC'])


# =============================================================================
class ColoredFormatter(logging.Formatter):
    # A variant of code found at:
    #  http://stackoverflow.com/questions/384076/how-can-i-make-the-python-logging-output-to-be-colored

    LEVEL_COLOR = {
        'DEBUG':    None,
        'INFO':     'GREEN',
        'WARNING':  'YELLOW',
        'ERROR':    ('BOLD', 'RED'),
        'CRITICAL': 'RED_BG',
    }

    # -------------------------------------------------------------------------
    def __init__(self, msg):

        logging.Formatter.__init__(self, msg)

    # -----------------------------------------------------------
    @property
    def color_debug(self):
        """The color used to output debug messages."""
        return self.LEVEL_COLOR['DEBUG']

    @color_debug.setter
    def color_debug(self, value):
        self.LEVEL_COLOR['DEBUG'] = value

    # -----------------------------------------------------------
    @property
    def color_info(self):
        """The color used to output info messages."""
        return self.LEVEL_COLOR['INFO']

    @color_info.setter
    def color_info(self, value):
        self.LEVEL_COLOR['INFO'] = value

    # -----------------------------------------------------------
    @property
    def color_warning(self):
        """The color used to output warning messages."""
        return self.LEVEL_COLOR['WARNING']

    @color_warning.setter
    def color_warning(self, value):
        self.LEVEL_COLOR['WARNING'] = value

    # -----------------------------------------------------------
    @property
    def color_error(self):
        """The color used to output error messages."""
        return self.LEVEL_COLOR['ERROR']

    @color_error.setter
    def color_error(self, value):
        self.LEVEL_COLOR['ERROR'] = value

    # -----------------------------------------------------------
    @property
    def color_critical(self):
        """The color used to output critical messages."""
        return self.LEVEL_COLOR['CRITICAL']

    @color_critical.setter
    def color_critical(self, value):
        self.LEVEL_COLOR['CRITICAL'] = value

    # -------------------------------------------------------------------------
    def format(self, record):

        record = copy.copy(record)
        levelname = record.levelname

        if levelname in self.LEVEL_COLOR:

            record.name = colorstr(record.name, 'BOLD')
            record.filename = colorstr(record.filename, 'BOLD')
            record.module = colorstr(record.module, 'BOLD')
            record.funcName = colorstr(record.funcName, 'BOLD')
            record.pathname = colorstr(record.pathname, 'BOLD')
            record.processName = colorstr(record.processName, 'BOLD')
            record.threadName = colorstr(record.threadName, 'BOLD')

            if self.LEVEL_COLOR[levelname] is not None:
                record.levelname = colorstr(levelname, self.LEVEL_COLOR[levelname])
                record.msg = colorstr(record.msg, self.LEVEL_COLOR[levelname])

        return logging.Formatter.format(self, record)

# =============================================================================

if __name__ == '__main__':

    pass

# =============================================================================

# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4
